esp-hal-common/src/soc/esp32/gpio.rs

```diff
@@ -72,10 +73,10 @@ pub(crate) fn get_io_mux_reg(gpio_num: u8) -> &'static crate::peripherals::io_mu
 }
 
 pub(crate) fn gpio_intr_enable(int_enable: bool, nmi_enable: bool) -> u8 {
-    int_enable as u8
-        | ((nmi_enable as u8) << 1)
-        | (int_enable as u8) << 2
-        | ((nmi_enable as u8) << 3)
+    match crate::get_core() {
```

Comment:
Just realized this was probably wrong even before - so worth double checking.

The TRM says
> GPIO_PINn_INT_ENA Interrupt enable bits for pin n: (R/W)
bit0: APP CPU interrupt enable;
bit1: APP CPU non-maskable interrupt enable;
bit3: PRO CPU interrupt enable;
bit4: PRO CPU non-maskable interrupt enable.

and also ESP-IDF says

> description: bit0: APP CPU interrupt enable  bit1: APP CPU non-maskable interrupt
 enable  bit3: PRO CPU interrupt enable  bit4:  PRO CPU non-maskable interrupt enable  bit5: SDIO's extent interrupt enable
 
However in tests everything looks fine 🤔

Reply:
Yeah this was weird, I initially implemented like it says in the TRM, but the logic was flipped, i.e setting the APP bits would allow interrupts to fire on the the PRO CPU. I'll take another look and see if I'm missing anything obvious.

Reply:
Actually,  I think it is correct as is but it's slightly weird because usually PRO CPU is mentioned first in the TRM but for some reason they decided to flip it around for this register :shrug: